source/blog/2023-11-14-hanami-210.html.markdown

```diff
@@ -0,0 +1,194 @@
+
+## It all starts with `hanami dev`
+
+Working on your app’s front end now starts with a single new command: `hanami dev`.
+
+Running `hanami dev` starts the familiar Hanami web server alongside our new **front end assets watcher and compiler**.
+
+From there, you’re ready to open `http://localhost:2300` and take in our gorgeous new welcome screen, in both light and dark mode.
+
+![Welcome screen in light mode](/blog/2023/11/14/hanami-210/welcome-light.png)
+
+![Welcome screen in dark mode](/blog/2023/11/14/hanami-210/welcome-dark.png)
+
+_Welcome (back!) to Hanami. We’ve been building something special for you!_
+
+## You’ll love our view on views
+
+To build your new front end, you can start with views. Like actions, **Hanami views are standalone, callable objects**, bringing a new level of clarity and reusability to the view layer.
+
+```ruby
+# app/views/posts/index.rb
+module MyApp
+  module Views
+    module Posts
+      class Index < MyApp::View
+        include Deps["posts_repo"]
+
+        expose :posts do |page:|
+          posts_repo.listing(page:)
+        end
+      end
+    end
+  end
+end
+```
+
+**View exposures** explicitly prepare the values we pass to templates, and they work seamlessly with Hanami’s Deps mixin, allowing you view to cleanly access other parts of your app as required.
+
+```
+<h1>Posts</h1>
+
+<%= posts.each do |post| %>
+  <h2><%= post.title %></h2>
+<% end %>
+```
+
+Hanami 2.1 delivers a **brand new ERB engine**, providing you a familiar template environment while also allowing for natural Ruby in view-focused methods, with a simple `yield` capturing nested template content, with no special handling required.
+
+Your views have access to **a library of familiar helpers**:
+
+```erb
+<%= form_for :post, routes.path(:create_post) do |f| %>
+  <%= f.label "title" %>
+  <%= f.text_field "title" %>
+<% end %>
+```
+
+You can write **your own helpers**, too:
+
+```ruby
+module MyApp
+  module Views
+    module Helpers
+      def warning_box
+        # tag is Hanami's built-in HTML builder helper
+        tag.div(class: "warning") do
+          yield # captures nested content in the template; so natural!
+        end
+      end
+    end
+  end
+end
+```
+
+Helpers can quickly become a mishmash, so Hanami provides **view parts** that encapsulate view logic right alongside the value it relates to. They can even render their own partials! This keeps your templates simple and allows you to use ordinary OO techniques to refactor and independently test more of your view code.
+
+```ruby
+module MyApp
+  module Views
+    module Parts
+      # Every post exposed from a view comes with these methods
+      class Post < MyApp::Views::Part
+        def title_link
+          helpers.tag.a(title, href: context.routes.path(:post, id:))
+        end
+
+        def feature_box
+          render("feature_box", title: title, text: teaser_text)
+        end
+      end
+    end
+  end
+end
+```
```

Comment:
It would be great to provide a simple ERB template example that shows the part, ie:

```erb
<ul>
  <% posts.each do |post| %>
    <li>
      <%= post.title_link %>
      <div>
        <%= post.feature_box %>
      </div>
    </li>
  <% end>
</ul>
```

Reply:
Done!